components/treelist/state.md

```diff
@@ -4,9 +4,875 @@ page_title: TreeList - State
+
+If you want to put the TreeList in a certain configuration without preserving the old one, create a `new TreeListState<T>()` and apply the settings there, then pass it to `SetState`.
+
+To reset the TreeList state, call `SetState(null)`.
+
+## Information in the TreeList State
+
+The following information is present in the TreeList state:
+
+* **Editing** - whether the user was inserting or editing an item (opens the same item for editing with the current data from the built-in editors of the TreeList - the data is updated in the `OnChange` event, not on every keystroke for performance reasons). The `OriginalEditItem` carries the original model without the user modifications so you can compare.
+
+* **Filtering** - filter descriptors (fields by which the Treelist is filtered, the operator and value).
+
+* **Paging** - page index
+
+* **Sorting** - sort descriptors (fields by which the TreeList is sorted, and the direction).
+
+* **Selection** - list of selected items.
+
+* **Columns** - Visible, Width, Index (order) of the column that the user sees, Locked (pinned).
+
+    * The TreeList matches the columns from its markup sequentially (in the same order) with the columns list in the state object. So, when you restore/set the state, the TreeList must initialize with the same collection of columns that were used to save the state.
+    
+        The `Index` field in the column state object represents its place (order) that the user sees and can choose through the `Reordable` feature, not its place in the TreeList markup. You can find an example below.
+    
+        If you want to change the visibility of columns, we recommend you use their `Visible` parameter rather than conditional markup - this parameter will be present in the state and will not change the columns collection count which makes it easier to reconcile changes.
+
+
+## Examples
+
+You can find the following examples in this section:
+
+* [Save and Load TreeList State from Browser LocalStorage](#save-and-load-treelist-state-from-browser-localstorage)
+* [Set TreeList Options Through State](#set-treelist-options-through-state)
+* [Set Default (Initial) State](#set-default-initial-state)
+* [Get The User Action That Changes The TreeList](#get-the-user-action-that-changes-the-treelist)
+* [Initiate Editing or Inserting of an Item](#initiate-editing-or-inserting-of-an-item)
+
+### Save and Load TreeList State from Browser LocalStorage
+
+The following example shows one way you can store the TreeList state - through a custom service that calls the browser's LocalStorage. You can use your own database here, or a file, or Microsoft's ProtectedBrowserStorage package, or any other storage you prefer. This is just an example you can use as base and modify to suit your project.
+  
+>note If you use Hierarchical data for the TreeList you need to serialize the current item only and not the entire collection in order not to exceed the size of the LocalStorage.
```

Comment:
Link the article about hierarchical data binding. Perhaps clarify a little "the entire collection" to something like "the entire collection of child Items"